src/WebJobs.Script/OutOfProc/Http/HttpFunctionInvokeDispatcher.cs

```diff
@@ -0,0 +1,200 @@
+
+namespace Microsoft.Azure.WebJobs.Script.OutOfProc
+{
+    internal class HttpFunctionInvokeDispatcher : IFunctionDispatcher
+    {
+        private readonly IMetricsLogger _metricsLogger;
+        private readonly ILogger _logger;
+        private readonly IHttpInvokerChannelFactory _httpInvokerChannelFactory;
+        private readonly IEnvironment _environment;
+        private readonly IScriptJobHostEnvironment _scriptJobHostEnvironment;
+        private readonly TimeSpan thresholdBetweenRestarts = TimeSpan.FromMinutes(LanguageWorkerConstants.WorkerRestartErrorIntervalThresholdInMinutes);
+
+        private IScriptEventManager _eventManager;
+        private IEnumerable<WorkerConfig> _workerConfigs;
+        private IDisposable _workerErrorSubscription;
+        private IDisposable _workerRestartSubscription;
+        private ScriptJobHostOptions _scriptOptions;
+        private bool _disposed = false;
+        private bool _disposing = false;
+        private IEnumerable<FunctionMetadata> _functions;
+        private ConcurrentStack<HttpWorkerErrorEvent> _languageWorkerErrors = new ConcurrentStack<HttpWorkerErrorEvent>();
+        private IHttpInvokerChannel _httpInvokerChannel;
+        private IHttpInvokerService _httpInvokerService;
+
+        public HttpFunctionInvokeDispatcher(IOptions<ScriptJobHostOptions> scriptHostOptions,
+            IMetricsLogger metricsLogger,
+            IEnvironment environment,
+            IScriptJobHostEnvironment scriptJobHostEnvironment,
+            IScriptEventManager eventManager,
+            ILoggerFactory loggerFactory,
+            IHttpInvokerChannelFactory httpInvokerChannelFactory,
+            IHttpInvokerService httpInvokerService,
+            IOptions<LanguageWorkerOptions> languageWorkerOptions,
+            IJobHostLanguageWorkerChannelManager jobHostLanguageWorkerChannelManager)
+        {
+            _metricsLogger = metricsLogger;
+            _scriptOptions = scriptHostOptions.Value;
+            _environment = environment;
+            _scriptJobHostEnvironment = scriptJobHostEnvironment;
+            _eventManager = eventManager;
+            _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
+            _logger = loggerFactory.CreateLogger<HttpFunctionInvokeDispatcher>();
+            _httpInvokerChannelFactory = httpInvokerChannelFactory;
+            _httpInvokerService = httpInvokerService;
+
+            State = FunctionDispatcherState.Default;
+
+            _workerErrorSubscription = _eventManager.OfType<HttpWorkerErrorEvent>()
+               .Subscribe(WorkerError);
+            _workerRestartSubscription = _eventManager.OfType<HttpWorkerRestartEvent>()
+               .Subscribe(WorkerRestart);
+        }
+
+        public FunctionDispatcherState State { get; private set; }
+
+        internal ConcurrentStack<HttpWorkerErrorEvent> LanguageWorkerErrors => _languageWorkerErrors;
```

Comment:
>LanguageWorkerErrors [](start = 55, length = 20)

Should these be called httpWorkerErrors or something? applies to JobHostLanguageWorkerChannel too.

Reply:
done. will clean up